### PR TITLE
Vault 2.4

### PR DESCRIPTION
**Pull request by ajm13**
General
- user storage limitation option
- file size limit increased to 1GB

Web UI
- dropzone properly handles multiple files dropped
- upload progress bar

Plugin
- sidebar with actions
- file sorting by name, size, and date
- "browse" export integration

### CODE REVIEW

File: nanome_vault/Vault.py [hunk 1]
```diff
@@ -142,7 +149,10 @@ def get_server_url(self):
             s.close()
 
         if port != DEFAULT_WEB_PORT:
-            url += ':' + str(port)
+            if port == 443:
```
Comment (mjrosengrant) [inline nanome_vault/Vault.py:152]: Could we add another port variable, similar to `DEFAULT_WEB_PORT`? Maybe `SECURE_WEB_PORT` (or a better name)

File: nanome_vault/Vault.py [hunk 2]
```diff
@@ -37,12 +37,16 @@ def on_run(self):
...
     @async_callback
-    async def on_export_file(self, request):
+    async def on_export_integration(self, request):
         await self.on_presenter_change()
         (location, filename, data) = request.get_args()
-        path = os.path.join(self.account, location)
-        self.vault.add_file(path, filename, data)
-        request.send_response(True)
+
+        if location == 'Browse':
```
Comment (mjrosengrant) [inline nanome_vault/Vault.py:44]: Would it be a lot of work to make EXPORT_LOCATIONS into an enum? I feel like this should look like

```
if location == EXPORT_LOCATIONS.BROWSE:
    ...
```

File: server/src/app.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ express.response.error = function (error, statusCode) {
 app.use(require('morgan')('dev')) // logging
 app.use(require('helmet')({ contentSecurityPolicy: false }))
 app.use(require('compression')())
-app.use(formidable({ multiples: true }))
+app.use(formidable({ multiples: true, maxFileSize: 1024 ** 3 }))
```
Comment (mjrosengrant) [inline server/src/app.js:28]: What does the ** 3 do?

Comment (ajm13) [reply server/src/app.js:28]: `**` is JS exponent operator, so 1024 * 1024 * 1024

**Pull request merged**